src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetRenameHelper.java

```diff
@@ -138,8 +143,32 @@ private int renameInScopeR()
                   return renameVariablesInScope(scope, targetValue, targetType);
          }
          
+         // If we're renaming variables within a chunk,
+         // apply the refactor across all R chunks.
          if (scope.isChunk())
-            return renameVariablesInScope(scope, targetValue, targetType);
+         {
+            JsArray<Scope> scopeTree = editor_.getScopeTree();
+            for (int i = 0, n = scopeTree.length(); i < n; i++)
+            {
+               if (scopeTree.get(i).isChunk())
+               {
+                  String headerLine = editor_.getLine(scopeTree.get(i).getPreamble().getRow());
+                  Map<String, String> chunkOptions = RChunkHeaderParser.parse(headerLine);
+                  String engine = chunkOptions.getOrDefault("engine", "r");
```

Comment:
We should probably provide a simpler API to get the chunk engine in this context, but this is workable.